ruth/simulator/kernels.py

```diff
@@ -31,22 +42,41 @@ def __init__(self, port: int):
         from ..zeromq.src.client import Client
         self.client = Client(port=port)
 
-    def compute_alternatives(self, vehicles: List[Vehicle], k: int) -> List[Optional[AlternativeRoutes]]:
-        od_paths = [(*v.next_routing_od_nodes, k) for v in vehicles]
-        inputs = [json.dumps(x).encode() for x in od_paths]
-        result = self.client.compute(inputs)
-        return result
+    def load_map(self, map: Map):
+        """
+        Loads updated information from the passed map.
+        """
+        # TODO: implement with broadcast
+        self.client.compute([Message(kind="load-map", data=map.hdf5_file_path)])
+
+    def compute_alternatives(self, map: Map, vehicles: List[Vehicle], k: int) -> List[
+        Optional[AlternativeRoutes]]:
+        messages = [Message(kind="alternatives", data={
+            "start": map.osm_to_hdf5_id(v.next_routing_od_nodes[0]),
+            "destination": map.osm_to_hdf5_id(v.next_routing_od_nodes[1]),
+            "max_routes": k
+        }) for v in vehicles]
+        # logging.info(f"Sending alternatives to distributed worker: {messages}")
+        results = self.client.compute(messages)
+        # logging.info(f"Response from worker: {results}")
+        remapped_routes = [
+            [[map.hdf5_to_osm_id(segment) for segment in route] for route in result["routes"]]
```

Comment:
Here segment should probably be named node_id instead.

Reply:
Is there something like an explicit segment ID (in the OSMNX maps or HDF5)? The code mentions it sometimes. Or do only nodes have IDs and "segment ID" is always just `node ID start - node ID end`?

Reply:
Yes, there is a segment ID, but the map doesn't use it for indexing, just in the segment data info section. The segments are indexed by start and end nodes.

Reply:
Ok, changed the naming :)